backend/docs/introducing_new_tools.md

```diff
@@ -0,0 +1,129 @@
+
+Adding new tool to MedTagger requires making changes both in backend and in frontend. To see how to add your tool on the frontend side proceed [here](). 
+
+### Step 1. Add tool to the definitions
+
+ Add your new tool to the `medtagger/definitions.py`. The `LabelTool` enum defines all tools supported by MedTagger.
+
+### Step 2. Create tool database model
+
+To enable support for various labels types we have introduced a concept of `LabelElement` which can be seen in `medtagger/database/models.py/`. It represents a high level LabelElement after which more specific types of LabelElement can inherit from. You can read more [here](http://docs.sqlalchemy.org/en/latest/orm/inheritance.html)
+
+As you can see from the `LabelElement` our `tool` column serves as the discriminator, which indicates the type of represented object:
+
+```python
+__mapper_args__ = {
+    'polymorphic_identity': 'LabelElement',
+    'polymorphic_on': tool,
+}
+```
+
+Now let's create our own model!
+
+```python
+class NewToolsLabelElement(LabelElement):
+    """Definition of a Label Element made with New Tool."""
+
+    __tablename__ = 'NewToolLabelElements'
+    id: LabelElementID = Column(String, ForeignKey('LabelElements.id'), primary_key=True)
+
+    # Here provide a set of parameters that are relevant to your NewToolLabelElement.
+
+    __mapper_args__ = {
+        'polymorphic_identity': LabelTool.NewTool,
+    }
+
+    # Your NewToolsLabelElement should also have __init__ and __repr__ methods.
+```
+
+### Step 3. Performing alembic migration
+
+After introducing new model to our database we need to update it. Please refer to [this](changing_database_models.md) tutorial to learn how to do that.
+
+### Step 4. Methods for adding new label
+
+#### Step 4.1 Repository
+
+Create a method in `medtagger/repositories/labels.py` that will create a new `db_session` and add your NewToolLabelElement. Please refer to other examples in the same file.
+
+#### Step 4.2 Business
+
+Create a method in `medtagger/api/scans/business.py` that will call method you have created in the last step. Please refer to other examples in the same file.
+
+Now search for `add_label_element` method in the same file. As you can see this method handless adding new label elements by the tool that the label was made with. Please add your tool and the method that will be responsible for creating such label.
+
+If your tool requires some additional validation. Please write the validation method and add it to `validate_label_payload` in `medtagger/api/scans/business.py`.
+
+#### Step 4.3 REST Service
+
+As you can see in `medtagger/api/scans/service_rest.py` we do already have endpoint that adds new label for given scan. What is now important is to define a schema that will ensure that whatever JSON representation of a label made with your tool is correct.
```

Comment:
(...) that will ensure that whatever JSON representation of a label made with your tool **is, it is correct**.